File: app/src/main/java/com/alexbelogurow/dbcoursework/Activity/ActivityAddPatient.java
package com.alexbelogurow.dbcoursework.Activity;

import android.os.Bundle;
import android.support.design.widget.CoordinatorLayout;
import android.support.design.widget.Snackbar;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

import com.alexbelogurow.dbcoursework.DataBase.DBHandler;
import com.alexbelogurow.dbcoursework.Model.Patient;
import com.alexbelogurow.dbcoursework.Model.Person;
import com.alexbelogurow.dbcoursework.R;

import java.util.ArrayList;
import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class ActivityAddPatient extends AppCompatActivity {

    private Toolbar mToolbar;
    private TextView mBirthDate;
    private List<CircleImageView> mImageViewListGender;
    private EditText mEditTextPatientName,
            mEditTextPatientLocation,
            mEditTextPatientJob,
            mEditTextPatientComments;
    private Spinner mSpinnerBloodType,
            mSpinnerRhFactor;

    private Button mButtonPickBirthDate,
            mButtonAddPatient;

    private CoordinatorLayout mCoordinatorLayout;

    private int numberOfCurImage = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_patient);

        initializeFields();


        setSupportActionBar(mToolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        mButtonPickBirthDate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DialogFragment dateDialog = new com.alexbelogurow.dbcoursework.Fragment.DatePicker();
                dateDialog.show(getSupportFragmentManager(), "datePicker");
            }
        });

        mButtonAddPatient.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Person person = new Person(
                        mEditTextPatientName.getText().toString(),
                        mBirthDate.getText().toString(),
                        numberOfCurImage == 0 ? "Male" : "Female");

                Patient patient = new Patient(
                        mSpinnerBloodType.getSelectedItem().toString(),
                        mSpinnerRhFactor.getSelectedItem().toString(),
                        mEditTextPatientLocation.getText().toString(),
                        mEditTextPatientJob.getText().toString(),
                        mEditTextPatientComments.getText().toString());

                Snackbar snackbar = Snackbar
                        .make(mCoordinatorLayout, "New Patient was added", Snackbar.LENGTH_SHORT);
                snackbar.show();

                DBHandler dbHandler = new DBHandler(getApplicationContext());

                dbHandler.addPatient(patient, person);
                Log.d(Log.DEBUG + "", dbHandler.getAllPatients().toString());
                dbHandler.close();
                snackbar.addCallback(new Snackbar.Callback() {
                    @Override
                    public void onDismissed(Snackbar transientBottomBar, int event) {
                        super.onDismissed(transientBottomBar, event);

                        onBackPressed();
                    }
                });

            }
        });
    }

    private void initializeFields() {
        mBirthDate = (TextView) findViewById(R.id.textViewPatientDate);
        mEditTextPatientName = (EditText) findViewById(R.id.editTextDoctorName);
        mEditTextPatientLocation = (EditText) findViewById(R.id.editTextPatientLocation);
        mEditTextPatientJob = (EditText) findViewById(R.id.editTextPatientJob);
        mEditTextPatientComments = (EditText) findViewById(R.id.editTextPatientComments);

        mButtonPickBirthDate = (Button) findViewById(R.id.buttonDoctorPickDate);
        mButtonAddPatient = (Button) findViewById(R.id.buttonAddPatient);

        mSpinnerBloodType = (Spinner) findViewById(R.id.spinnerBloodType);
        mSpinnerRhFactor = (Spinner) findViewById(R.id.spinnerRhFactor);

        mImageViewListGender = new ArrayList<>();
        mImageViewListGender.add((CircleImageView) findViewById(R.id.imageViewDoctorMale));
        mImageViewListGender.add((CircleImageView) findViewById(R.id.imageViewDoctorFemale));

        mImageViewListGender.get(0).setTag(0);
        mImageViewListGender.get(1).setTag(1);

        mToolbar = (Toolbar) findViewById(R.id.toolbarAddPatient);
        mCoordinatorLayout = (CoordinatorLayout) findViewById(R.id.coordLayoutAddPatient);
    }

    public void onClickImage(View view) {
        int numberOfClicked = (int) view.getTag();
        if (numberOfCurImage != numberOfClicked) {
            numberOfCurImage = numberOfClicked;

            int numberOfSecondImage = Math.abs(numberOfCurImage - 1);

            if (numberOfCurImage == 0) {
                mImageViewListGender.get(0).setImageResource(R.drawable.ic_patient_male_on);
                mImageViewListGender.get(1).setImageResource(R.drawable.ic_patient_female_off);

            }

            if (numberOfCurImage == 1) {
                mImageViewListGender.get(1).setImageResource(R.drawable.ic_patient_female_on);
                mImageViewListGender.get(0).setImageResource(R.drawable.ic_patient_male_off);
            }

            mImageViewListGender.get(numberOfCurImage).
                    setBorderColorResource(R.color.colorBorderOn);
            mImageViewListGender.get(numberOfSecondImage).
                    setBorderColorResource(R.color.colorBorderOff);


        }
    }


}

File: app/src/main/java/com/alexbelogurow/dbcoursework/Activity/ActivityAddPatient.kt
package com.alexbelogurow.dbcoursework.Activity

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.CoordinatorLayout
import android.support.design.widget.Snackbar
import android.support.v4.app.DialogFragment
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView

import com.alexbelogurow.dbcoursework.DataBase.DBHandler
import com.alexbelogurow.dbcoursework.Model.Patient
import com.alexbelogurow.dbcoursework.Model.Person
import com.alexbelogurow.dbcoursework.R

import java.util.ArrayList

import de.hdodenhof.circleimageview.CircleImageView

class ActivityAddPatient : AppCompatActivity() {

    private var EXTRA_PATIENT_ID = "PATIENT_ID"

    private var mToolbar: Toolbar? = null
    private var mBirthDate: TextView? = null
    private var mImageViewListGender: MutableList<CircleImageView>? = null
    private var mEditTextPatientName: EditText? = null
    private var mEditTextPatientLocation: EditText? = null
    private var mEditTextPatientJob: EditText? = null
    private var mEditTextPatientComments: EditText? = null
    private var mSpinnerBloodType: Spinner? = null
    private var mSpinnerRhFactor: Spinner? = null

    private var mButtonPickBirthDate: Button? = null
    private var mButtonAddDoctorForPatient: Button? = null

    private var mCoordinatorLayout: CoordinatorLayout? = null

    private var numberOfCurImage = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_patient)

        initializeFields()

        setSupportActionBar(mToolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        mButtonPickBirthDate?.setOnClickListener {
            val dateDialog = com.alexbelogurow.dbcoursework.Fragment.DatePicker()
            dateDialog.show(supportFragmentManager, "datePicker")
        }

        mButtonAddDoctorForPatient?.setOnClickListener {
            val person = Person(
                    mEditTextPatientName?.text.toString(),
                    mBirthDate?.text.toString(),
                    if (numberOfCurImage == 0) "Male" else "Female")

            val patient = Patient(
                    mSpinnerBloodType?.selectedItem.toString(),
                    mSpinnerRhFactor?.selectedItem.toString(),
                    mEditTextPatientLocation?.text.toString(),
                    mEditTextPatientJob?.text.toString(),
                    mEditTextPatientComments?.text.toString())

            val dbHandler = DBHandler(applicationContext)
            val patientID = dbHandler.addPatient(patient, person)
            dbHandler.close();

            val intentAddDoctor = Intent(this, ActivityAddDoctorForPatient::class.java)
            intentAddDoctor.putExtra(EXTRA_PATIENT_ID, patientID)
            startActivity(intentAddDoctor)
        }
        /*
        mButtonAddPatient!!.setOnClickListener {
            val person = Person(
                    mEditTextPatientName!!.text.toString(),
                    mBirthDate!!.text.toString(),
                    if (numberOfCurImage == 0) "Male" else "Female")

            val patient = Patient(
                    mSpinnerBloodType!!.selectedItem.toString(),
                    mSpinnerRhFactor!!.selectedItem.toString(),
                    mEditTextPatientLocation!!.text.toString(),
                    mEditTextPatientJob!!.text.toString(),
                    mEditTextPatientComments!!.text.toString())

            val snackbar = Snackbar
                    .make(mCoordinatorLayout!!, "New Patient was added", Snackbar.LENGTH_SHORT)
            snackbar.show()

            val dbHandler = DBHandler(applicationContext)

            dbHandler.addPatient(patient, person)
            Log.d(Log.DEBUG.toString() + "", dbHandler.allPatients.toString())
            dbHandler.close()
            snackbar.addCallback(object : Snackbar.Callback() {
                override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
                    super.onDismissed(transientBottomBar, event)

                    onBackPressed()
                }
            })
        }
        */

    }

    private fun initializeFields() {
        mBirthDate = findViewById<View>(R.id.textViewPatientDate) as TextView
        mEditTextPatientName = findViewById<View>(R.id.editTextDoctorName) as EditText
        mEditTextPatientLocation = findViewById<View>(R.id.editTextPatientLocation) as EditText
        mEditTextPatientJob = findViewById<View>(R.id.editTextPatientJob) as EditText
        mEditTextPatientComments = findViewById<View>(R.id.editTextPatientComments) as EditText

        mButtonPickBirthDate = findViewById<View>(R.id.buttonDoctorPickDate) as Button
        mButtonAddDoctorForPatient = findViewById(R.id.button_add_doctor_for_patient)

        mSpinnerBloodType = findViewById<View>(R.id.spinnerBloodType) as Spinner
        mSpinnerRhFactor = findViewById<View>(R.id.spinnerRhFactor) as Spinner

        mImageViewListGender = ArrayList()
        mImageViewListGender!!.add(findViewById<View>(R.id.imageViewDoctorMale) as CircleImageView)
        mImageViewListGender!!.add(findViewById<View>(R.id.imageViewDoctorFemale) as CircleImageView)

        mImageViewListGender!![0].tag = 0
        mImageViewListGender!![1].tag = 1

        mToolbar = findViewById<View>(R.id.toolbarAddPatient) as Toolbar
        mCoordinatorLayout = findViewById<View>(R.id.coordLayoutAddPatient) as CoordinatorLayout
    }

    fun onClickImage(view: View) {
        val numberOfClicked = view.tag as Int
        if (numberOfCurImage != numberOfClicked) {
            numberOfCurImage = numberOfClicked

            val numberOfSecondImage = Math.abs(numberOfCurImage - 1)

            if (numberOfCurImage == 0) {
                mImageViewListGender!![0].setImageResource(R.drawable.ic_patient_male_on)
                mImageViewListGender!![1].setImageResource(R.drawable.ic_patient_female_off)

            }

            if (numberOfCurImage == 1) {
                mImageViewListGender!![1].setImageResource(R.drawable.ic_patient_female_on)
                mImageViewListGender!![0].setImageResource(R.drawable.ic_patient_male_off)
            }

            mImageViewListGender!![numberOfCurImage].setBorderColorResource(R.color.colorBorderOn)
            mImageViewListGender!![numberOfSecondImage].setBorderColorResource(R.color.colorBorderOff)


        }
    }

}
